### PR TITLE
Newly Added files for Wifi SDK 3.0 Integration

### PR DESCRIPTION
**Pull request by sharad-patil24**
Created a Separate PR for newly added files for Wifi SDK 3.0 Integration and Common Flash.

This PR is in continuation of below PR
https://github.com/project-chip/connectedhomeip/pull/28558

### CODE REVIEW

File: src/platform/silabs/rs911x/ble_config.h [hunk 1]
```diff
@@ -0,0 +1,272 @@
+/*******************************************************************************
```
Comment (jepenven-silabs) [inline src/platform/silabs/rs911x/ble_config.h:1]: This files should be removed all together. It litterally changes 7 defines that could easily be encapsulated inside an ifdef (in rsi_ble_config.h)

File: examples/platform/silabs/SiWx917/device/src/startup_common_RS1xxxx.c [hunk 2]
```diff
@@ -0,0 +1,512 @@
+/**
+ ******************************************************************************
```
Comment (jepenven-silabs) [inline examples/platform/silabs/SiWx917/device/src/startup_common_RS1xxxx.c:2]: Should be move to Matter support

File: examples/platform/silabs/SiWx917/SiWx917/sl_wlan_config.h [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+#endif
+                     .tcp_ip_feature_bit_map = (SL_SI91X_TCP_IP_FEAT_DHCPV4_CLIENT | SL_SI91X_TCP_IP_FEAT_DNS_CLIENT |
+                                                SL_SI91X_TCP_IP_FEAT_SSL | SL_SI91X_TCP_IP_FEAT_BYPASS
+#ifdef ipv6_FEATURE_REQUIRED
+                                                | SL_SI91X_TCP_IP_FEAT_DHCPV6_CLIENT | SL_SI91X_TCP_IP_FEAT_IPV6
+#endif
+                                                | SL_SI91X_TCP_IP_FEAT_ICMP | SL_SI91X_TCP_IP_FEAT_EXTENSION_VALID),
+                     .custom_feature_bit_map     = (SL_SI91X_FEAT_CUSTOM_FEAT_EXTENTION_VALID | RSI_CUSTOM_FEATURE_BIT_MAP),
+                     .ext_custom_feature_bit_map = (
+#ifdef CHIP_917
```
Comment (jepenven-silabs) [inline examples/platform/silabs/SiWx917/SiWx917/sl_wlan_config.h:54]: Is this defines from the WiseConnect SDK ?

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c [hunk 4]
```diff
@@ -0,0 +1,796 @@
...
+#include "event_groups.h"
+#include "sl_board_configuration.h"
+#include "sl_net.h"
+#include "sl_si91x_host_interface.h"
+#include "sl_si91x_types.h"
+#include "sl_wifi_callback_framework.h"
+#include "sl_wifi_constants.h"
+#include "sl_wifi_types.h"
+#include "sl_wlan_config.h"
+#include "task.h"
```
Comment (jepenven-silabs) [inline examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c:35]: this looks a lot like the `rsi_wifi_if.c` already present in the repo. Maybe some optimization could be done (pure virtual call and per board implementation or something of the sorts)

File: examples/platform/silabs/SiWx917/SiWx917/sl_wlan_config.h (hunk 3 above)
Comment (sharad-patil24) [reply examples/platform/silabs/SiWx917/SiWx917/sl_wlan_config.h:54]: No these defines are present in New WIFI SDK as well

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c [hunk 5]
```diff
@@ -0,0 +1,797 @@
...
+                        hasNotifiedWifiConnectivity = true;
+                    }
+                }
+                else if (dhcp_state == DHCP_OFF)
+                {
+                    wfx_ip_changed_notify(IP_STATUS_FAIL);
+                    hasNotifiedIPV4 = false;
+                }
+#endif /* CHIP_DEVICE_CONFIG_ENABLE_IPV4 */
+                /*
```
Comment (rosahay-silabs) [inline examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c:606]: Remove line 606-613

File: src/platform/silabs/rs911x/ble_config.h (hunk 1 above)
Comment (sharad-patil24) [reply src/platform/silabs/rs911x/ble_config.h:1]: this file has dependency in SDK , SDK is expecting a file with this name

../../../examples/lock-app/silabs/third_party/connectedhomeip/third_party/silabs/wifi_sdk/components/si91x/ble/src/rsi_ble_gatt_apis.c:18:10: fatal error: ble_config.h: No such file or directory
   18 | #include "ble_config.h"

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c [hunk 6]
```diff
@@ -603,14 +603,6 @@ void wfx_rsi_task(void * arg)
                     hasNotifiedIPV4 = false;
                 }
 #endif /* CHIP_DEVICE_CONFIG_ENABLE_IPV4 */
-                /*
```
Comment (jepenven-silabs) [inline examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c]: @sharad-patil24 this should be removed and added to another PR. It doesn't have anything to do in this PR

Comment (jepenven-silabs) [reply examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c]: I see this is for the new file. Will leave it as is for now.

**Review by jepenven-silabs — APPROVED** [review]

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c (hunk 5 above)
Comment (jepenven-silabs) [reply examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c]: Seems to be removed

**Review by andy31415 — APPROVED** [review]
